core/src/main/scala/kafka/log/Log.scala

```diff
@@ -337,6 +337,9 @@ class Log(@volatile private var _dir: File,
         }
     } else if (keepPartitionMetadataFile) {
       _topicId.foreach(partitionMetadataFile.write)
+    } else {
```

Comment:
This was a pre-existing issue but should we extract this whole if else block to a method? It's also a little bit easy to miss that it's setting the `_topicid` in https://github.com/apache/kafka/pull/10754/files#diff-eeafed82ed6a8600c397b108787fdf31e03191b0a192774a65c127d0d26edc44L330. Maybe that could benefit from an empty line prior to it.

Reply:
Yeah. Unfortunately there is a lot going on here. I've edited the comment and spacing to make it a little clearer. If we still think it needs adjustment, I can try making a new method.

Reply:
I think extracting it into a method named initializeTopicId or something like that would be good.